### PR TITLE
Add BigLake API Table Resource

### PR DESCRIPTION
**Pull request by nevzheng**
Add BigLake API Table Resource

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_biglake_table`
```

### CODE REVIEW

**Comment by nevzheng** [comment]
All PRs For BigLake APIs

github.com/GoogleCloudPlatform/magic-modules/pull/8751
github.com/GoogleCloudPlatform/magic-modules/pull/8753
github.com/GoogleCloudPlatform/magic-modules/pull/8754

**Comment by nevzheng** [comment]
note Provider File is in Catalog PR: go/magic-modules/pull/8751

I learned you can't generate the documentation without this file.

**Review by melinath — COMMENTED** [review]
commenting to remove from my review queue for now since there are some dependencies for this PR - let me know when I should take a look

**Comment by nevzheng** [comment]
Hi Stephan,
We merged Catalog. go/magic-modules/pull/8751 this provides us with the product.yaml file.

If we were confused about the difference between "BigLake Metastore API" (this PR) and "BigLake Managed Tables" (#8722): These are different services with similar names. Metastore is GA, and we can proceed with merging this CL into `google-provider` GA.

While BLMT is in experimental, and we're discussing whether EAP is more appropriate

Best
nevin

**Comment by nevzheng** [comment]
oh wait. We need to wait for go/magic-modules/pull/8753. sorry.

File: mmv1/products/bigquerybiglake/Table.yaml
Comment (melinath) [inline mmv1/products/bigquerybiglake/Table.yaml:55]: generally for child resources like this, instead of all of these parameters, we would have a `database` field that is expected to contain everything up through `databaseId`. So for example, the `base_url` would become:

```yaml
base_url: "{{database}}/tables"
```

This allows users to more easily directly reference the parent resource. For example:

```hcl
resource "google_biglake_database" "database" {
  // ...
}
resource "google_biglake_table" "table" {
  database = google_biglake_database.database.id
}
```

**Review by melinath — CHANGES_REQUESTED** [review]
At a quick glance this all seems reasonable except one thing (mentioned below). I'll take a closer look once the remaining blocking PR is merged. Thanks!

File: mmv1/products/bigquerybiglake/Table.yaml
Comment (nevzheng) [reply mmv1/products/bigquerybiglake/Table.yaml:55]: ack for now.

I did not follow this convention in the other PRs, which were merged. What should I do? See #8751 #8753

Comment (melinath) [reply mmv1/products/bigquerybiglake/Table.yaml:55]: Catalog isn't a child resource, so it's fine. I've opened https://github.com/GoogleCloudPlatform/magic-modules/pull/8790 to update the implementation of the Database resource. It would be great to make similar changes here.

**Review by melinath — CHANGES_REQUESTED** [review]
Note that we've also changed the mmv1 folder name to `biglake` to match the product API endpoint, so you'll need to move `Table.yaml` to that folder and rename the example to `biglake_table`.

**Comment by nevzheng** [comment]
Blocked on #8790

File: mmv1/products/biglake/Table.yaml
Comment (melinath) [inline mmv1/products/biglake/Table.yaml:29]: Instead of using a custom_import, it should be possible to use an import_format that includes a `%` in the parent field name. basically the same change as I ended up making here: https://github.com/GoogleCloudPlatform/magic-modules/pull/8790/commits/f91638889ce1920976b7ee1b8413c52856965b3e

File: mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go
Comment (melinath) [inline mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go:82]:
```suggestion
```

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/products/biglake/Table.yaml
Comment (nevzheng) [reply mmv1/products/biglake/Table.yaml:29]: done.

File: mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go
Comment (nevzheng) [reply mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go:70]: done.

File: mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go
Comment (nevzheng) [reply mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go:82]: done.

**Comment by nevzheng** [comment]
```
 + serde_info {                          + serialization_lib = "org.apache.hadoop.hive.serde2.avro.AvroSerDe"                        }
```

is giving me some trouble. cross checking with team...

**Review by melinath — CHANGES_REQUESTED** [review]
> ```
>  + serde_info {                          + serialization_lib = "org.apache.hadoop.hive.serde2.avro.AvroSerDe"                        }
> ```
> 
> is giving me some trouble. cross checking with team...

Acknowledged, removing from my review queue for now.

**Comment by nevzheng** [comment]
Update: There's a backend bug. `serde_info` is not handled in the backend. I'm working on fixing it. I'll ping @melinath when we're ready. it might take a while due to rollouts.

**Comment by melinath** [comment]
@nevzheng would it make sense to remove that field from this PR and then add it in a separate PR after the rollout completes?

**Comment by nevzheng** [comment]
SGTM. will update PR.

**Comment by nevzheng** [comment]
Hi Stephen,
as discussed:
1. remove serde_info
2. fix backend
3. rollout change
4. submit followup PR.

i'll keep track of the CI run, but am expecting no issues, atm.

**Comment by nevzheng** [comment]
repeating. the following field is not supported in the api due an oversight, and we are working on implementing it.

```
 serde_info {
        serialization_lib = # value needed
      }
```

File: mmv1/products/biglake/Table.yaml
Comment (melinath) [inline mmv1/products/biglake/Table.yaml:135]: @nevzheng it looks like the field still needs to be removed here.

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/products/biglake/Table.yaml
Comment (nevzheng) [reply mmv1/products/biglake/Table.yaml]: done.

File: mmv1/templates/terraform/examples/biglake_table.tf.erb
Comment (melinath) [inline mmv1/templates/terraform/examples/biglake_table.tf.erb:55]: Not a blocker but it could be nice to follow the guidance at https://developers.google.com/style/examples#example-person-names to choose a name.

File: mmv1/products/biglake/Table.yaml
Comment (melinath) [inline mmv1/products/biglake/Table.yaml:50]: just want to double-check that this is mutable? Changing a parent field usually requires deleting and recreating a resource.

File: mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go
Comment (melinath) [inline mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go:84]: It could be good to have the update tests alter table_type and storage_descriptor subfields but that's not necessarily a blocker.

**Review by melinath — COMMENTED** [review]
LGTM - tests are complete & passing, and the resource definition looks good. Just one quick question.

File: mmv1/templates/terraform/examples/biglake_table.tf.erb
Comment (nevzheng) [reply mmv1/templates/terraform/examples/biglake_table.tf.erb:55]: done.

File: mmv1/products/biglake/Table.yaml
Comment (nevzheng) [reply mmv1/products/biglake/Table.yaml:50]: done.
marked immutable

File: mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go
Comment (nevzheng) [reply mmv1/third_party/terraform/services/biglake/resource_biglake_table_test.go:84]: done.

updated fields in test.

**Review by melinath — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
